core/src/main/java/com/adobe/aem/commons/assetshare/util/UrlUtil.java

```diff
@@ -52,14 +56,48 @@ public static final String escape(final String unescaped, final boolean preventD
         }
 
         String tmp = unescaped;
+        String host = null;
+        String queryParams = null;
+
+        // Handle hosts
+        int indexOfScheme = StringUtils.indexOf(tmp, "://");
```

Comment:
Why aren't you using a [URL](https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#URL(java.lang.String)) object to breakdown the string, rather than parsing it yourself?